plugins/action_update_review_status/src/updateReviewsStatuses.ts

```diff
@@ -18,38 +18,96 @@ const updateReviewsStatuses: ActionPluginType = async ({
   const stageId = parameters?.stageId || applicationData?.stageId
-  const currentReviewLevel = applicationData?.reviewData?.levelNumber || 0
-  const level = parameters?.level || currentReviewLevel + 1
+  const currentReviewLevel = parameters.level || applicationData?.reviewData?.levelNumber || 0
+
+  if (isReview) console.log('Review submitted id', reviewId, 'level ', currentReviewLevel)
+  else console.log('Application submitted id', applicationId)
+
+  console.log('Changed responses', changedResponses)
 
   const reviewsToUpdate = []
 
-  console.log('changedApplicationResponses', changedApplicationResponses)
+  const getReviewsByLevelAndStatus = async (
+    level: number,
+    statusToUpdate: ReviewStatus[]
+  ): Promise<Review[]> =>
+    (await db.getAssociatedReviews(applicationId, stageId, level)).filter(
+      (review: Review) =>
+        review.reviewId !== reviewId && statusToUpdate.includes(review.reviewStatus)
+    )
+
+  console.log('Finding reviews to update status...')
 
   try {
-    // Get reviews/review assignments (with status) matching application_id & current stage
-    const reviews = (
-      await db.getAssociatedReviews(applicationId, stageId, level)
-    ).filter((review: Review) =>
-      [ReviewStatus.Submitted, ReviewStatus.Locked, ReviewStatus.Draft].includes(
-        review.reviewStatus
-      )
-    )
-    // Deduce which ones should be updated
-    for (const review of reviews) {
-      const { reviewAssignmentId, levelNumber, reviewStatus } = review
-      if (levelNumber > 1) reviewsToUpdate.push({ ...review, reviewStatus: ReviewStatus.Pending })
-      else if (await haveAssignedResponsesChanged(reviewAssignmentId))
-        reviewsToUpdate.push({ ...review, reviewStatus: ReviewStatus.Pending })
-      else if (reviewStatus === ReviewStatus.Locked)
-        reviewsToUpdate.push({ ...review, reviewStatus: ReviewStatus.Pending })
+    if (isReview) {
+      // Review submitted from upper level to lower level review
+      if (decision === Decision.ChangesRequested) {
+        const previousLevelReview = currentReviewLevel - 1
+        // Update lower level reviews with assigned responses on the array changed responses ...
+        const reviewsSubmitted = await getReviewsByLevelAndStatus(previousLevelReview, [
+          ReviewStatus.Submitted,
+        ])
+        // Deduce which ones should be updated to Pending
+        for (const review of reviewsSubmitted) {
+          const { reviewAssignmentId } = review
+          if (await haveAssignedResponsesChanged(reviewAssignmentId))
+            reviewsToUpdate.push({ ...review, reviewStatus: ReviewStatus.ChangesRequested })
+        }
+      } else {
+        // Review submitted from lower level to upper level
+        // Update same level reviews as submitted if no other still have changes requested status...
+        const othersChangeRequested = await getReviewsByLevelAndStatus(currentReviewLevel, [
```

Comment:
hmm, I am not sure about the logic in this bit I think you describe it here:

> For same-level reviews, when LOCKED they will change to DRAFT (or kept as LOCKED if other CHANGES_REQUESTED reviews in the same level)

I think it's not needed. locked reviews will be changed to pending when applicant re-submits application, and the only time they will be locked is when consolidator (lvl + 1) submits a decision other then `change request`

Reply:
Ok, I can remove this

Reply:
Thanks, that be great

Reply:
Hi, moving back to this. I though your request was just to remove the logic to check if other CHANGES_REQUEST have been done before changing from LOCKED to DRAFT. 

I agree in removing this for with test and then we can work on a more general solution on issue #364 : )